DashChat.cs

```diff
@@ -342,21 +344,44 @@ public void ChangeState(DSState changeTo)
         chatState = changeTo;
         OnChangedState?.Invoke(changeTo);
     }
+
     //Handles variables inside of line of text
     string HandleVariable(string _variableKey)
     {
-        object match = regexVariable.Match(_variableKey);
+        Match match = regexVariable.Match(_variableKey);
+        string providedValue = string.Empty;
+        string variableName = match.Groups[1].Value;
+
+        if (VariableProvider != null)
+        {
+            providedValue = VariableProvider(variableName);
+        }
+        else if (OnLookUpVariable != null)
+        {
+            //consider passing variableName instead of full match (tag).
+            providedValue = EventBasedVariableLookup(match.ToString());
+        }
+
+        return regexVariable.Replace(_variableKey, providedValue);
+    }
+
+    [Obsolete(message: "Assign variable provider delegate to DashChat.VariableValueProvider instead.")]
+    private string EventBasedVariableLookup(string match)
+    {
+        string providedValue = string.Empty;
+
```

Comment:
Lets just drop this all together now since no one other than us are playing with the script no sense of supporting a deprecation warning.